packages/hash/frontend/src/pages/[account-slug]/entities/[entity-uuid].page/entity-editor/links-section/link-table/cells/linked-with-cell/linked-with-cell-editor.tsx

```diff
@@ -0,0 +1,170 @@
+  const { archiveEntity } = useBlockProtocolArchiveEntity();
+  const { aggregateEntities } = useBlockProtocolAggregateEntities();
+
+  const { value: cell, onFinishedEditing } = props;
+  const { expectedEntityTypes, linkAndTargetEntities, linkEntityTypeId } =
+    cell.data.linkRow;
+
+  const [search, setSearch] = useState("");
+
+  const [entities, setEntities] = useState<Entity[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [open, setOpen] = useState(false);
+  const highlightedRef = useRef<null | Entity>(null);
+  const router = useRouter();
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        setLoading(true);
+        const { data } = await aggregateEntities({
+          data: {
+            rootEntityTypeIds: expectedEntityTypes.map(
+              ({ schema }) => schema.$id,
+            ),
+          },
+        });
+
+        if (data) {
+          setEntities(getRoots(data));
+        }
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    void init();
+  }, [aggregateEntities, expectedEntityTypes]);
+
+  const onCreateNew = () => {
+    if (!expectedEntityTypes[0]) {
+      return;
+    }
+
+    const accountSlug = router.query["account-slug"];
+    /** @todo ask about this behavior */
+    void window.open(
+      `/${accountSlug}/entities/new?entity-type-id=${encodeURIComponent(
+        expectedEntityTypes[0].schema.$id,
+      )}`,
+      "_blank",
+    );
+  };
+
+  const onSelect = async (val: Entity) => {
+    const { linkEntity: currentLink, rightEntity: currentLinkedEntity } =
+      linkAndTargetEntities[0] ?? {};
+
+    const sameEntity =
+      currentLinkedEntity?.metadata.editionId.baseId ===
+      val.metadata.editionId.baseId;
+
+    // if clicked on the same entity, do nothing
+    if (sameEntity) {
+      return onFinishedEditing();
+    }
+
+    // if there is an existing link, archive it
+    if (currentLink) {
+      await archiveEntity({
+        data: { entityId: currentLink.metadata.editionId.baseId },
+      });
+    }
+
+    // create new link
+    await createEntity({
+      data: {
+        entityTypeId: linkEntityTypeId,
+        properties: {},
+        linkMetadata: {
+          leftEntityId: getRoots(entitySubgraph)[0]?.metadata.editionId.baseId!,
+          rightEntityId: val.metadata.editionId.baseId,
+        },
+      },
+    });
+
+    await refetch();
+    onFinishedEditing(undefined);
+  };
+
+  const onCancel = () => {
+    onFinishedEditing();
+  };
+
+  return (
+    <HashSelectorAutocomplete
+      className="click-outside-ignore"
+      dropdownProps={{
+        query: search,
+        createButtonProps: {
+          className: "click-outside-ignore",
+          onMouseDown: (evt) => {
+            evt.preventDefault();
+            evt.stopPropagation();
+            onCreateNew();
+          },
+        },
+        variant: "entity",
+      }}
+      loading={loading}
+      options={entities.sort((a, b) =>
```

Comment:
`.sort` mutates, and `entities` is in React state. You need to wrap this with `[...entities]`

Reply:
Oops missed this, thanks!